### ISSUE TITLE
validate terminal values and provide errors

### ISSUE THREAD

**Issue by mess110** [dilcom/pilot-gnuplot#2]
While experimenting with the samples, I noticed I was missing certain terminals. To see a list of available terminals:

```
gnuplot> set terminal
```

It would be nice if this could be done from ruby. We could use a function which returns a list of enabled terminals, and, when a terminal is configured, validate the values. Example:

```
Plot.new([...], title: 'foo', term: ['invalid']) # expect to raise error
```

There could be many invalid gnuplot syntax errors. Maybe we can take care of some of the common ones.

**Comment by ievgrafov** [comment]
@mess110 Terminal validation done. It is validated in Plot#new, Plot#plot (raise ArgumentError) and Plot#method_missing. So now call something like Plot#to_unexisting_term will raise NoMethodError.

I also think about validating other options and left a place where this may be added.

**Comment by mess110** [comment]
I tested this out and it works fine, thank you.

I also saw you added a version check. Nice.

Yes, I agree with you we can validate other options as well. We do however, need to be careful not to validate something which can change with a new gnuplot release. For example:

Instead of calling 'set term', we could have used a constant array. This would have been a **bad idea** because you would have to maintain that array, when gnuplot changes. And have other bad side effects (like not solving our problem with installed terminals)
